backends/occa/ceed-occa-qfunction.cpp

```diff
@@ -183,8 +183,10 @@ namespace ceed {
     int QFunction::getContextData() {
       if (qFunctionContext) {
         int ierr;
+        void *data = NULL;
         ierr = CeedQFunctionContextGetData(qFunctionContext, CEED_MEM_DEVICE,
-                                           &qFunctionContextData);
+                                           &data);
+        qFunctionContextData = &data;
         CeedOccaFromChk(ierr);
       }
       return 0;
```

Comment:
My work is failing because I'm not able to update this variable for some strange reason. I must be missing something about how OCCA memory works.

But this highlights a serious concern I have with this backend. In this code, I am attempting to access the CeedQFunctionContext object's device memory via the backend interface function specifically designed to do so. However, as a general rule this backend bypasses the libCEED backend interface, preferring to directly pass data between backend specific objects. 

Since this backend is designed to bypass the libCEED backend interface, this means that this backend skips error checking we placed in that interface, this backend is not interoperable with other backends (which means I cannot provide reference implementations of preconditioning ingredients), and some libCEED features such as memory state tracking are not available to users who will use this backend.

At this point, we're locked into this approach of bypassing the libCEED backend interface in this overhaul, but we should take note of the drawbacks and increased maintenance burden from this design.